### ISSUE TITLE
Should we give admins or managers the ability to delete the project

### ISSUE THREAD

**Issue by vikasrohit** [appirio-tech/connect-app#2296]
@rmjmccormick @acshields @vic-topcoder we don't have this capability yet in the application, do you want to have this in place to remove the spam projects?

**Comment by rmjmccormick** [comment]
@vikasrohit @acshields @vic-topcoder Define spam projects please? Do you mean "test" projects set up in prod? If so, we should identify these and delete through the back-end. I'd be very wary of granting full delete powers to a wider audience. We can discuss the merits tomorrow though.

**Comment by vikasrohit** [comment]
@rmjmccormick Spam projects I am referring to are test projects created by us (prod dev) or any topcoder user which is marked as spam/demo by operations team. We get many spam projects daily in our system.
and when it comes to deleted them from backend, that is tricky because we have to deleted many entities associated with a project when a project is deleted. We can allow only `administrators` or `Connect Admin` to be able to delete the projects.
Also, fyi @rmjmccormick @acshields @vic-topcoder, till date we are having hard delete for projects in connect i.e. once a customer deletes a project, we can never retrieve it back even if it was a mistake. We have enabled soft delete for all project related entities in 2.4.4 release, so from now on, if a project is got deleted by mistake we have the opportunity to get it back.

**Comment by vic-topcoder** [comment]
I think in the past the decision was made to not delete spam projects, but hide them from the interface. Those are the "draft" projects and in general with the upcoming updates to project creation the friction to create a project would increase and this would limit the number of spam. I agree we do not want to start deleting projects form the database. We're not at the point where we have to start saving on storage or recycle project IDs.

Just out of curiosity - what would be the value of deleting spam projects? Are there any benefits to the performance of the system?

**Comment by vikasrohit** [comment]
> Just out of curiosity - what would be the value of deleting spam projects? Are there any benefits to the performance of the system?

Yes, down the road they would definitely have significant impact on performance considering that very high percentage of our total projects are spam projects.

**Comment by vikasrohit** [comment]
@vic-topcoder @acshields Do we have requirements finalised?
